core/src/main/java/io/grpc/ServiceDescriptor.java

```diff
@@ -204,4 +205,19 @@ public ServiceDescriptor build() {
       return new ServiceDescriptor(this);
     }
   }
+
+  @Override
+  public String toString() {
+    String toString =  MoreObjects.toStringHelper(this)
+        .add("name", name)
+        .add("schemaDescriptor", schemaDescriptor)
+        .omitNullValues()
+        .toString();
+
+    for (MethodDescriptor<?, ?> methodDescriptor : methods) {
+      toString += "\n\t" + methodDescriptor.toString();
```

Comment:
I looked at some of the other `toString` methods in gRPC and they all seem to be a single line. I'd prefer to keep this a single line as well and optimize for grepping at the cost of human readability.